controllers/vspheremachine_controller.go

```diff
@@ -239,25 +239,29 @@ func (r *machineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 
 	// Checking whether cluster is nil here as we still want to allow delete even if cluster is not found.
 	if cluster == nil {
+		log.Info("Cluster could not be fetched")
 		return reconcile.Result{}, nil
 	}
 
+	if cluster.Spec.InfrastructureRef == nil {
+		log.Info("Cluster.spec.infrastructureRef is not yet set")
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
q: why are we using requeing after here? if I'm not wrong this controller is watching for clusters, so it will reconcile again when InfrastructureRef is set.

Reply:
No, I don't think so. Should be fine.